docs/states.md

```diff
@@ -0,0 +1,53 @@
+| `offer_withdrawn`            | Provider withdraws an offer |
+| `pending_conditions`         | Provider makes and offer and the caniddate must meet conditions of the offer |
+| `recruited`                  | Unconditional offer is accepted or offer conditions have been met |
+| `rejected`                   | Application is rejected by provider accompanied by structured reasons|
+| `unsubmitted`                | Candidate has added the course to their application form but has not submitted it yet |
+| `withdrawn`                  | Submitted application has been withdrawn by the candidate (or by the provider on behalf of the canidate) |
+
+## State Categories
+
+These are groupings of states that allow us to define behaviour for a any application whose status is in the group.
+
+|Category|States|Description|
+|---|----|---|
+|STATES_NOT_VISIBLE_TO_PROVIDER |unsubmitted<br>cancelled<br>application_not_sent|These states provide functionality in the system but are not exposed to the Providers. They are either hidden completely from the API or aliased to another state name|
+|STATES_VISIBLE_TO_PROVIDER|awaiting_provider_decision<br>interviewing<br>offer<br>pending_conditions<br>recruited<br>rejected<br>declined<br>withdrawn<br>conditions_not_met<br>offer_withdrawn<br>offer_deferred<br>inactive|Theoretically the opposite of VISIBLE_TO_PROVIDER|
+|INTERVIEWABLE_STATES|awaiting_provider_decision<br>interviewing<br>inactive|States in which an application can have an interview created for them|
+|ACCEPTED_STATES|pending_conditions<br>conditions_not_met<br>recruited<br>offer_deferred|The states that are possible once a provider has made an offer and the candidate has accepted.|
+|OFFERED_STATES|(ACCEPTED_STATES + declined offer offer_withdrawn)|All States possible if a provider wants to offer a candidate a place|
+|POST_OFFERED_STATES|(ACCEPTED_STATES + declined offer_withdrawn)|All States possible from the offer state|
+|UNSUCCESSFUL_STATES|withdrawn<br>cancelled<br>rejected<br>declined<br>conditions_not_met<br>offer_withdrawn<br>application_not_sent<br>inactive|All states that end in a state that is not recruited|
+|SPENT_STATES|withdrawn<br>cancelled<br>rejected<br>declined<br>conditions_not_met<br>offer_withdrawn<br>application_not_sent<br>inactive|States of all applications that do not contribute to the limit of applications a candidate can have concurrently on their application form|
+|SUCCESSFUL_STATES|pending_conditions<br>offer<br>offer_deferred<br>recruited||
+|DECISION_PENDING_STATUSES|awaiting_provider_decision<br>interviewing||
+|DECISION_PENDING_AND_INACTIVE_STATUSES|awaiting_provider_decision<br>interviewing<br>inactive||
+|REAPPLY_STATUSES|rejected<br>cancelled<br>withdrawn<br>declined<br>offer_withdrawn||
+|TERMINAL_STATES|UNSUCCESSFUL_STATES<br>recruited|Used only in ApplicationMonitor (SupportInterface)|
+|IN_PROGRESS_STATES|DECISION_PENDING_STATUSES<br>ACCEPTED_STATES<br>offer|Used only in application_form.rb:314|
```

Comment:
A link to the line where it's used could be useful as the line will change

Reply:
Good suggestion, I'm going to remove that reference now though and give a description of the category instead of saying where it's used.